### PR TITLE
[AWS]: Add Secrets Manager

### PR DESCRIPTION
**Pull request by crisboarna**
## Description
This introduces a new command which covers another common use case for developers using AWS: verifying/copying secrets variables.

- Add "Secrets Manager" to find, view, and copy a secret value

## Screencast

image
image
image
image

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/amazon-aws/src/secrets.tsx [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+              <List.Item.Detail.Metadata.Label title="Last Accessed" text={secret.LastAccessedDate?.toLocaleString()} />
+              <List.Item.Detail.Metadata.Separator />
+              <List.Item.Detail.Metadata.Label title="Rotation" text={`${secret.RotationEnabled ? "true" : "false"}`} />
+            </List.Item.Detail.Metadata>
+          }
+        />
+      }
+      actions={
+        <ActionPanel>
+          <Action.SubmitForm title={showValue ? "Hide Value" : "Show Value"} onSubmit={() => setShowValue()} />
```
Comment (JonathanWbn) [inline extensions/amazon-aws/src/secrets.tsx:78]: I'm aware other commands have used the `SubmitForm` action for such a use case, but a simple [Action](https://developers.raycast.com/api-reference/user-interface/actions#action) is probably a better choice.

```suggestion
          <Action title={showValue ? "Hide Value" : "Show Value"} onSubmit={() => setShowValue()} />
```

**Review by JonathanWbn — CHANGES_REQUESTED** [review]
Awesome new command, great job! 🎉  I really like the approach you chose with the `Detail` panel. 👏 

I have two minor tweaks to suggest.

File: extensions/amazon-aws/src/secrets.tsx [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+          markdown={secretDetails?.SecretString || secretDetails?.SecretBinary?.toString() || ""}
+          metadata={
+            <List.Item.Detail.Metadata>
+              <List.Item.Detail.Metadata.Label title="Description" text={secret.Description} />
+              <List.Item.Detail.Metadata.Label title="ARN" text={secret.ARN} />
+              <List.Item.Detail.Metadata.Separator />
+              <List.Item.Detail.Metadata.Label title="Creation Date" text={secret.CreatedDate?.toLocaleString()} />
+              <List.Item.Detail.Metadata.Label title="Last Accessed" text={secret.LastAccessedDate?.toLocaleString()} />
+              <List.Item.Detail.Metadata.Separator />
+              <List.Item.Detail.Metadata.Label title="Rotation" text={`${secret.RotationEnabled ? "true" : "false"}`} />
```
[the remark this answers is not in the document]
Comment (crisboarna) [reply extensions/amazon-aws/src/secrets.tsx]: Fixed

File: extensions/amazon-aws/src/secrets.tsx (hunk 1 above)
Comment (crisboarna) [reply extensions/amazon-aws/src/secrets.tsx]: Fixed

**Comment by crisboarna** [comment]
Hi @JonathanWbn,

Could you have a look once more if it is good to go ?

**Comment by JonathanWbn** [comment]
I am wondering if using Raycast's built-in filtering is actually the better choice for the secrets command. I introduced server-side filtering for parameters and log groups because there might be thousands or even tens of thousands of them. I'm not too familiar with the secrets manager, but I would assume there are rarely thousands of secrets per profile. If this is the case, I would suggest removing the search approach for a more straight-forward implementation and better UX. What do you think, @crisboarna?

**Comment by crisboarna** [comment]
Hi @JonathanWbn,

I think that in most scenarios server side filtering is the best way to go as it 

1. Increases speed of commands by reducing payload network traffic
2. Reduces amount of work needed to be done on front-end/raycast
3. Reduces memory footprint overall

This is valid if there are 10 or 1000 secrets. 

I personally have profiles with several hundred secrets for larger projects and Secrets Manager supports up to 500,000 secrets per region so there is the potential to have a lot of secrets and it is best to code for worst case scenarion.

TL;DR: I think server side rendering is best way to go for this as well :)

**Review by JonathanWbn — APPROVED** [review]

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥
